akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -130,12 +131,11 @@ import akka.stream.scaladsl.Source
       val session = CassandraSessionRegistry(system).sessionFor(sessionConfigPath)
 
       val offsetStore = new CassandraOffsetStore(session)
-
       val readOffsets = () => offsetStore.readOffset(projectionId)
 
       val source: Source[(Offset, Envelope), NotUsed] =
         Source
-          .futureSource(sourceProvider.source(readOffsets))
+          .futureSource(tryStartHandler().flatMap(_ => sourceProvider.source(readOffsets)))
```

Comment:
The handler is start with the source, that's good. But the stop handler is called from the projection api own stop method. 

What happens in case of a stream failure and the backoff restart it? We need to hook the handle stop in the stream failure.

Reply:
oh, disregard that comment. I'm wrong on this.

Reply:
hmm, it's not quite wrong what I said. :-)

The stream Future (when completed) is used to fabricate a new Future using the stop handler Future, but the backoff won't see it. It will restart the source and we will have a race between the new start and the stop handler.

Reply:
After looking into it in more details, here is what I think it will happen:

The failure never leaks out of the backoff source so there is no stream Future that get's completed, unless explicitly stopped by the ProjectionBehavior or a gracefully completed stream.

The backoff will restart the handler again after each stream failure.

Reply:
Good point. I have to look into that. Either it should only start/stop once or start/stop for each restart.

Reply:
The restart will use the same handler instance, so I think that starting it once is correct. 

It makes the implementation on our side much simpler as well.

Reply:
I think it's necessary to call start/stop also for the restarts. Otherwise the handler will not know that it should reload or reinitialize after failure. Indeed more difficult, but I have implemented it in e2f0687.

Also found a bug in the restart withBackoff. It never actually restarted since it returned same Source again (not reading latest offset).